File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EOnSecondaryFixedVariantDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnSecondaryFixedVariantDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd32
            AppTheme.STANDARD -> R.color.cstd32
            AppTheme.BLUE -> R.color.cbd32
            AppTheme.GREEN -> R.color.cgd32
            AppTheme.RED -> R.color.crd32
            AppTheme.NOTHING -> R.color.cnod32
        }
    }
}

File: lib/layer2_core/dark/e_on_secondary_fixed_variant_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EOnSecondaryFixedVariantDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd32;
      case 'standard': return cstd32;
      case 'blue': return cbd32;
      case 'green': return cgd32;
      case 'red': return crd32;
      case 'nothing': return cnod32;
      default: return cstd32;
    }
  }
}
